starknet_devnet/blueprints/rpc/structures/payloads.py

```diff
@@ -0,0 +1,390 @@
+class FunctionCall(TypedDict):
+    """
+    TypedDict for rpc function call
+    """
+    contract_address: Address
+    entry_point_selector: Felt
+    calldata: List[Felt]
+
+
+def rpc_invoke_transaction(transaction: InvokeSpecificInfo) -> RpcInvokeTransaction:
+    """
+    Convert gateway invoke transaction to rpc format
+    """
+    txn: RpcInvokeTransaction = {
+        "contract_address": rpc_felt(transaction.contract_address),
+        "entry_point_selector": rpc_felt(transaction.entry_point_selector),
+        "calldata": [rpc_felt(data) for data in transaction.calldata],
+        "transaction_hash": rpc_felt(transaction.transaction_hash),
+        "max_fee": rpc_felt(transaction.max_fee),
+        "version": hex(0x0),
+        "signature": [rpc_felt(value) for value in transaction.signature],
+        "nonce": rpc_felt(0),
+        "type": rpc_txn_type(transaction.tx_type.name),
+    }
+    return txn
+
+
+def rpc_declare_transaction(transaction: DeclareSpecificInfo) -> RpcDeclareTransaction:
+    """
+    Convert gateway declare transaction to rpc format
+    """
+    txn: RpcDeclareTransaction = {
+        "class_hash": rpc_felt(transaction.class_hash),
+        "sender_address": rpc_felt(transaction.sender_address),
+        "transaction_hash": rpc_felt(transaction.transaction_hash),
+        "max_fee": rpc_felt(transaction.max_fee),
+        "version": hex(transaction.version),
+        "signature": [rpc_felt(value) for value in transaction.signature],
+        "nonce": rpc_felt(transaction.nonce),
+        "type": rpc_txn_type(transaction.tx_type.name),
+    }
+    return txn
+
+
+def rpc_deploy_transaction(transaction: DeploySpecificInfo) -> RpcDeployTransaction:
+    """
+    Convert gateway deploy transaction to rpc format
+    """
+    txn: RpcDeployTransaction = {
+        "transaction_hash": rpc_felt(transaction.transaction_hash),
+        "class_hash": rpc_felt(transaction.class_hash),
+        "version": hex(0x0),
+        "type": rpc_txn_type(transaction.tx_type.name),
+        "contract_address": rpc_felt(transaction.contract_address),
+        "contract_address_salt": rpc_felt(transaction.contract_address_salt),
+        "constructor_calldata": [rpc_felt(data) for data in transaction.constructor_calldata],
+    }
+    return txn
+
+
+class RpcFeeEstimate(TypedDict):
+    """
+    Fee estimate TypedDict for rpc
+    """
+    gas_consumed: NumAsHex
+    gas_price: NumAsHex
+    overall_fee: NumAsHex
+
+
+def rpc_fee_estimate(fee_estimate: dict) -> dict:
+    """
+    Convert gateway estimate_fee response to rpc_fee_estimate
+    """
+    result: RpcFeeEstimate = {
+        "gas_consumed": hex(fee_estimate["gas_usage"]),
+        "gas_price": hex(fee_estimate["gas_price"]),
+        "overall_fee": hex(fee_estimate["overall_fee"]),
+    }
+    return result
+
+
+def make_invoke_function(request_body: dict) -> InvokeFunction:
+    """
+    Convert RPC request to internal InvokeFunction
+    """
+    return InvokeFunction(
```

Comment:
I don't think it can be used here since every field is converted to int, with some fields having ternary operators like `max_fee=int(request_body["max_fee"], 16) if "max_fee" in request_body else 0`